### ISSUE TITLE
javascript error in console leading to perpetual wait for search results in alignment pipeline

### ISSUE THREAD

**Issue by adf-ncgr** [legumeinfo/gcv#420]
A seemingly innocuous request:
https://gcv.legumeinfo.org/gene;lis=cicar.CDCFrontier.gnm1.ann1.Ca_21954?algorithm=repeat&match=10&mismatch=-1&gap=-1&score=30&threshold=25&bmatched=20&bintermediate=10&bmask=10&linkage=average&cthreshold=20&neighbors=50&matched=4&intermediate=5&sources=lis&bregexp=&border=chromosome&regexp=&order=distance
leads to console error:
main.c67d416a58e79492.js:1 ERROR RangeError: Invalid array length
and somewhat misleading pipeline status:
<img width="636" alt="image" src="https://user-images.githubusercontent.com/2009605/161988653-7b642c80-d6c2-4501-ada9-c89292c8fd82.png">
which leaves the user in some perplexity.

**Comment by adf-ncgr** [comment]
this may be a less innocuous request but leads to the same apparent behavior and derives from one of the example searches on the gcv.legumeinfo.org welcome page:
https://gcv.legumeinfo.org/gene;lis=medtr.A17.gnm5.ann1_6.MtrunA17Chr3g0078741?q=medtr.A17.gnm5.MtrunA17Chr3:1000000-1500000&sources=lis&algorithm=repeat&match=10&mismatch=-1&gap=-1&score=30&threshold=25&bmatched=20&bintermediate=10&bmask=10&linkage=average&cthreshold=20&neighbors=70&matched=4&intermediate=5&bregexp=&border=chromosome&regexp=&order=distance

**Comment by alancleary** [comment]
Thanks for bumping this issue. It got buried among the other issues! Unless it's a high priority, this probably won't get fixed until June when I'm back from vacation. It's worth noting that switching the alignment algorithm to Smith-Waterman prevents the error: https://gcv.legumeinfo.org/gene;lis=medtr.A17.gnm5.ann1_6.MtrunA17Chr3g0078741?q=medtr.A17.gnm5.MtrunA17Chr3:1000000-1500000&sources=lis&algorithm=smith-waterman&match=10&mismatch=-1&gap=-1&score=30&threshold=25&bmatched=20&bintermediate=10&bmask=10&linkage=average&cthreshold=20&neighbors=70&matched=4&intermediate=5&bregexp=&border=chromosome&regexp=&order=distance

**Comment by adf-ncgr** [comment]
thanks, not urgent I just happened to run into again when doing something else. Good to know it is only an issue for the repeat algorithm, I guess that should help narrow down the cause later.

**Comment by alancleary** [comment]
> Good to know it is only an issue for the repeat algorithm, I guess that should help narrow down the cause later.

Yeah, that's not at all surprising. Stitching together all the forward and reverse repeat alignment fragments is overly complicated at the moment. Hopefully addressing this issue helps simplify the code!
